nvflare/app_common/abstract/statistics_spec.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, List
+
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.statistics.stats_def import Feature, Histogram
```

Comment:
should these "Feature" "Histogram" be in this file as well? Since this file is the "spec"
Or maybe move stats_def inside the abstract folder?

Reply:
I just feel that the spec importing things from implementation is weird that's why I suggest that.